clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsTest.java

```diff
@@ -1170,7 +1169,7 @@ public void testSliceEmptyRecords() {
      */
     @ParameterizedTest
     @ArgumentsSource(MemoryRecordsArgumentsProvider.class)
-    public void testSliceForAlreadySlicedMemoryRecords(Args args) throws IOException {
+    public void testSliceForAlreadySlicedMemoryRecords(Args args) {
```

Comment:
Can we also remove explicit casting in this class i.e. at line 1092, 1093, etc. The similar we removed in FileRecordsTest.

Reply:
Thanks! I searched the entire repo for casts to MemoryRecord and FileRecords and I think we got all of them.